joint_calling/upload_processor.py

```diff
@@ -0,0 +1,110 @@
+""" This function prepares gVCF's uploaded to GCS, based off sample status
+    logged in a database, for further QC and downstream analysis.
+    The upload processor will determine when samples should be added
+    to existing MatrixTables where appropriate and which MatrixTables
+    they should be combined with in this case. Following a successful
+    run all uploaded files will be moved to archival storage.
+
+    Assumptions
+    ===========
+    - Specified files must exist in either the source or the destination
+      bucket
+    - Source and destination bucket must exist
+    - User must be authenticated with appropriate permissions """
+
+import errno
+import os
+import subprocess
+from typing import List
+import hailtop.batch as hb
+
+
+def batch_move_files(
+    batch: hb.batch,
+    files: List[str],
+    source_prefix: str,
+    destination_prefix: str,
+    docker_image: str = None,
+    key: str = None,
+) -> List:
+    """This script takes a Hail.Batch workflow and a list of files to
+    move between buckets. It adds 1 job per file to the workflow, and
+    runs in a docker_image. It returns the list of created jobs. Each
+    job consists of a gsutil move command for each valid file.
+    When run, these jobs will perform the batch move file operation.
+
+    Parameters
+    ==========
+    batch: hb.Batch
+        An object representing the DAG of jobs to run.
+    files: List[str]
+        A list of the file names to be moved.
+        For example ["TOB1543.g.vcf.gz","TOB2314.g.vcf.gz","TOB3423.g.vcf.gz"]
+    source_prefix: str
+        The path to the sub-directory where the files are initially located.
+        For example "cpg-tob-wgs-upload" or "cpg-tob-wgs-upload/v1"
+    destination_prefix: str
+        The path to the sub-directory where the files should be moved.
+        For example "cpg-tob-wgs-main" or "cpg-tob-wgs-upload/batch0"
+    docker_image: str, optional
```

Comment:
You could use the `Optional` type to make these semantics known to the type checker.

Reply:
Thanks! Applied in latest commit :)